extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -137,6 +138,16 @@ export class AmpSelector extends AMP.BaseElement {
         this.toggle_(args['index'], args['value']);
       }
     }, ActionTrust.LOW);
+
+    // Listening to this event
+    // Simply listening for mutation would not work
+    this.element.addEventListener(AmpEvents.DOM_UPDATE, unusedEvent => {
+      // Clear prev states
```

Comment:
to prevent unnecessary work, we should check if existing `this.options_` array is different than `querySelectorAll('[option]')`, if the same, break. Would be a nice util to add in `array.js` ( `areEqual(arr1, arr2)` ) .

This is important to do because majority of `amp-list`s or other components inside `amp-selector` won't be touching `options` and are there to render random other content unrelated to options. This is common because `amp-selector` is uses in cases such as creating `tabs` components where each tab itself hosts random content.